### PR TITLE
Room refactor

### PR DESCRIPTION
**Pull request by fado**

### CODE REVIEW

File: game_data.py [hunk 1]
```diff
@@ -1,12 +1,41 @@
...
-        "description": "You're standing outside a tavern. It's raining.",
-        "exits": {"inside": "Tavern"},
-    }
-}
+from lib.models.entity import Room
+from lib.models.enums import ExitType
 
+rooms = {}
+
+def add_room(room: Room):
```
Comment (fichilleen) [inline game_data.py:6]: What's your thinking with this function? It's no different that manually calling the dict

File: lib/models/entity.py [hunk 2]
```diff
@@ -14,6 +18,61 @@ def __init__(self, name: str, description: str, damage: int):
...
+            self.aliases = aliases
+            self.description = description
+
+
+    def __init__(self, 
+                 name: str, 
+                 description: str,
+                 description_items: List[DescriptionItem] = None,
+                 exits: List[Exit] = None,
+                 light_level: LightLevel = LightLevel.BRIGHT,
```
Comment (fichilleen) [inline lib/models/entity.py:46]: Did you forget to add a file where these are defined?

File: lib/models/entity.py [hunk 3]
```diff
@@ -14,6 +18,61 @@ def __init__(self, name: str, description: str, damage: int):
         super().__init__(name, description)
 
 
+class Room(Entity):
+
+    class Exit(Entity):
```
Comment (fichilleen) [inline lib/models/entity.py:23]: Why make these subclasses? It means when you're defining them, or referring to them, you have to access them as `Room.Exit`

It also means the class _definition_ is redefined every time a `Room` is instantiated. We're not at the point where these types of performance issues matter, but it's worth knowing about at least

File: game_data.py [hunk 4]
```diff
@@ -1,12 +1,41 @@
...
-    },
-    "Outside": {
-        "description": "You're standing outside a tavern. It's raining.",
-        "exits": {"inside": "Tavern"},
-    }
-}
+from lib.models.entity import Room
+from lib.models.enums import ExitType
 
+rooms = {}
```
Comment (fichilleen) [inline game_data.py:4]: Did you consider moving to an eg `Rooms` class, to get this out of the global namespace?

**Review by fichilleen — APPROVED** [review]
Generally good, but I think you might be going a little too far with some of the definitions here, in that it feels premature to be writing code for lighting levels, etc.

https://en.wikipedia.org/wiki/YAGNI

File: lib/models/entity.py (hunk 2 above)
Comment (fado) [reply lib/models/entity.py:46]: `lib.constants`

File: game_data.py (hunk 4 above)
Comment (fado) [reply game_data.py:4]: I will probably do that in the next iteration.

File: lib/models/entity.py (hunk 3 above)
Comment (fado) [reply lib/models/entity.py:23]: Fixed.

File: game_data.py (hunk 1 above)
Comment (fado) [reply game_data.py:6]: I think it's just a wee bit nicer than

`rooms['tavern'] = Room(name = 'tavern'...`

Less chance of human error.

**Review by fado — COMMENTED** [review]
I think generally you're right.  But I'll leave it for now.

**Pull request merged**